include/redis.h

```diff
@@ -88,15 +88,49 @@ class Redis : public RedisServer
         */
         Redis& operator=(Redis&& cluster) = default;
 
+       /*!
+        *   \brief Run a single-key Command on the server
+        *   \param cmd The single-key Comand to run
+        *   \returns The CommandReply from the
+        *            command execution
+        */
+        virtual CommandReply run(SingleKeyCommand& cmd);
+
         /*!
-        *   \brief Run a single-key or single-hash slot
-        *          Command on the server
-        *   \param cmd The single-key or single-hash
-        *              slot Comand to run
+        *   \brief Run a multi-key Command on the server
+        *   \param cmd The multi-key Comand to run
         *   \returns The CommandReply from the
         *            command execution
         */
-        virtual CommandReply run(Command& cmd);
+        virtual CommandReply run(MultiKeyCommand& cmd);
+
+        /*!
+        *   \brief Run a compound Command on the server
+        *   \param cmd The compound Comand to run
+        *   \returns The CommandReply from the
+        *            command execution
+        */
+        virtual CommandReply run(CompoundCommand& cmd);
+
+        /*!
+        *   \brief Run a non-keyed Command that addresses
+        *          the given db node(s) on the server
+        *   \param cmd The non-keyed Command that
+        *              addresses the given db node(s)
```

Comment:
Maybe just make the doc strings singular ``db node`` since the object only has one address and port?